mailbox/tools/indexer/src/main/java/org/apache/mailbox/tools/indexer/ReIndexerPerformer.java

```diff
@@ -69,32 +70,32 @@ public ReIndexerPerformer(MailboxManager mailboxManager,
         this.mailboxSessionMapperFactory = mailboxSessionMapperFactory;
     }
 
-    Mono<Result> reIndex(MailboxId mailboxId, ReprocessingContext reprocessingContext) {
+    Mono<Result> reIndex(MailboxId mailboxId, ReprocessingContext reprocessingContext, RunningOptions runningOptions) {
         MailboxSession mailboxSession = mailboxManager.createSystemSession(RE_INDEXER_PERFORMER_USER);
 
         return mailboxSessionMapperFactory.getMailboxMapper(mailboxSession)
             .findMailboxByIdReactive(mailboxId)
-            .flatMap(mailbox -> reIndex(reprocessingContext, mailboxSession, mailbox));
+            .flatMap(mailbox -> reIndex(reprocessingContext, mailboxSession, mailbox, runningOptions));
```

Comment:
This throttling is wrong:

 - We need, from the mailboxId, to generate a `Flux<Message>` that we then throttle.

Nesting operation would lead to the throttling being applied per mailbox, what is nlikely not what you want.